Makefile

```diff
@@ -10,19 +10,26 @@ BUILDFLAGS := -tags "$(BUILDTAGS)"
 
 PACKAGES := $(shell go list ./... | grep -v vendor)
 
-all: deps .gitvalidation test validate
+all: deps.timestamp .gitvalidation test validate
 
-deps: vndr
+vndr:
+	@go get -u github.com/LK4D4/vndr
+
+deps: vndr vendor deps.timestamp
 	go get -u $(BUILDFLAGS) github.com/golang/lint/golint
 	go get $(BUILDFLAGS) github.com/vbatts/git-validation
 
-vndr: vendor.conf
-	@go get -u github.com/LK4D4/vndr && vndr
+deps.timestamp:
+	@touch deps.timestamp
+
+vendor: deps.timestamp vendor.conf
+	@vndr
+	@touch vendor
 
 clean:
-	rm -rf vendor
+	rm -rf vendor deps.timestamp
 
-test: deps
+test: deps.timestamp
```

Comment:
I don't think that's going to work; IIRC make won't trigger on directory updates.